python/cudf/cudf/utils/utils.py

```diff
@@ -489,39 +489,64 @@ def get_appropriate_dispatched_func(
         if cupy_func is func:
             return NotImplemented
 
-        cupy_compatible_args = get_cupy_compatible_args(args)
-        cupy_output = cupy_func(*cupy_compatible_args, **kwargs)
-        return cast_to_appropriate_cudf_type(cupy_output)
+        cupy_compatible_args, index = _get_cupy_compatible_args_index(args)
+        if cupy_compatible_args:
+            cupy_output = cupy_func(*cupy_compatible_args, **kwargs)
+            return _cast_to_appropriate_cudf_type(cupy_output, index)
+        else:
+            return NotImplemented
```

Comment:
Same as below, I think you can just dump to the outer return

Reply:
Resolved by https://github.com/rapidsai/cudf/pull/6839/commits/665234136db75eb1b1024e125bff853df6cf27a1